### PR TITLE
HW1

### PR DESCRIPTION
**Pull request by nikitz77**
Testing for push from pycharm

### CODE REVIEW

**Comment by nikitz77** [comment]
Code found here

File: HW1-KZaballa.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import plotly.graph_objects as go
+from sklearn.datasets import make_classification
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import Normalizer
+from sklearn.svm import SVC
+
+# Read in data files
+iris_df = pd.read_csv("D:/SDSU BDA/BDA 696/HW1/HW1-Data/iris.data.csv")
```
Comment (dafrenchyman) [inline HW1-KZaballa.py:16]: Not the correct way to load the dataset since it'll use the first row as the header. Also, I obviously don't have this folder on my computer, so the code didn't run when I tried it.

File: HW1-KZaballa.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+# iris_df1 = pd.read_csv("D:/SDSU BDA/BDA 696/HW1/HW1-Data/iris.names.csv")
+iris_df.head
+
+# Descriptive Statistics
+iris_df.describe()
```
Comment (dafrenchyman) [inline HW1-KZaballa.py:21]: This should be wrapped in a print, so it goes to screen.

File: HW1-KZaballa.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+# Normalizer does not accept qualitative data
+X = iris_df[["Sepal Length", "Sepal Width", "Petal Length", "Petal Width"]]
+X_trans = Normalizer().transform(X)
+# set norm to mean of sepal length and transform
+# it by that
+#
+# Is my logic correct here? I am thinking in
+# order to truly normalize something, each max or mean
+# or whatever function has to be applied to each column
+# transformer = Normalizer(norm=max(X["Sepal Length"]),
```
Comment (dafrenchyman) [inline HW1-KZaballa.py:142]: You should look at the documentation if you want to change the normal behavior of the normalizer. It was just a silly function I wanted you guys to use,.. not really useful really.

File: HW1-KZaballa.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    marginal_x="histogram",
+    marginal_y="rug",
+    color="Class",
+)
+fig4.show()
+
+#   6) Different scatter-line plot styles
+#
+iris_df.head
+fig5 = go.Figure()
```
Comment (dafrenchyman) [inline HW1-KZaballa.py:93]: I forget which plots it were, but some just really didn't make sense.

**Review by dafrenchyman — APPROVED** [review]
First off, let’s start with the bad. You did catch that the header was missing from the dataset, but you lost one row of data because the first row was used as the header. While you did do the plots, some of them made no sense. I didn’t know what was being represented. You also didn’t use the `main` function. It may not seem like much, but when things get more complicated, it’s a life saver. You also didn’t put print statements around the output, so I didn’t see what the results were. I’m assuming you wrote this all in a jupyter notebook and then copied it over. It did run (after I had to fix where the data was located, I think this is related to jupyter as well), but it didn’t output the results to screen. As for your questions, the normalizer is just so you to make you have a transform only step, it’s really not that important, and you can’t call it the way you were trying. You could look at the documentation (I provided a link on the assignment page). Now for the good: The main part of this assignment was to get all the stuff setup: the linting, the the package requirements, etc. You set all of that up correctly. Your code was also easy to follow along, and although it didn’t print the results I wanted to screen, you at least had them in there as comments (I gave you a few extra points for that to negate the loss of points from the missing print statements).
